### ISSUE TITLE
Event order conflict between q-input buttons and blur event

### ISSUE THREAD

**Issue by dsl101** [quasarframework/quasar#1256]
### Software version

Quasar: 0.14.8
OS: Windows 10
Node: 8.6.0
NPM: 5.3.0
Browsers:
iOS:
Android:
Any other software related to your bug:

### What did you get as the error?
When combining a 'Save' icon and a blur event in a `q-input`, the blur event prevents the icon click handler from being called

### What were you expecting?
The click handler to be called

### What steps did you take, to get the error?
I'm using this setup on the `q-input`:

      <q-input v-if="adding"
        clearable
        ref="inputBox"
        v-model.lazy.trim="newData"
        :placeholder="'Add ' + adding"
        @keydown.enter="handleAdd"
        @blur="$nextTick(() => { newData = adding = null })"
        :after="[
          {
            icon: 'check',
            content: true,
            handler: handleAdd
          }
        ]"
      />

The idea is that the input is hidden, and when activated (by `adding` being set to some value), it appears. The 'save' icon should trigger `handleAdd()`, but the `@blur` event happens first (should clicking an icon _inside_ a `q-input` really blur the input?). Even so, I was hoping that using `$nextTick()` would allow the `handleAdd()` function to run first, but it seems that event is cancelled completely.

**Comment by pdanpdan** [comment]
Hello.

1. In the current version there is indeed a problem with loosing focus when clicking on attached buttons.
In the next version (quasar-future/v0.15) this is fixed.

2. Watch out that for the moment v-model does not accept .lazy modifier for components (there is a PR https://github.com/vuejs/vue/pull/6940 waiting for integration)
Meanwhile you can use the expanded form of
```
:value="newData" @change="val => { newData = typeof val === 'string' ? val.trim() : '' }"
```

But from what I understand you should use http://quasar-framework.org/components/chips-input.html

**Comment by dsl101** [comment]
Thanks for the info, and great that it's fixed. I'll be moving to 0.15 asap.

It looks like #882 this might be related too, but I'm not sure.

And no, it's an input that it being used to capture 3 different kinds of free format text (not chips) depending on which button the user clicks. The full code is like this, and it works really nicely, apart from the blur issue:

    <q-card-actions v-if="can">
      <span v-if="!adding">
        <q-btn v-for="section in sections" :key="section.key"
          outline
          rounded
          no-caps
          color="primary"
          @click="add(section.key)"
        >{{section.buttonLabel}}</q-btn>
      </span>
      <q-input v-if="adding"
        clearable
        ref="inputBox"
        class="full-width"
        v-model.lazy.trim="newData"
        :placeholder="'Add ' + adding"
        @keydown.enter="handleAdd"
        @blur="$nextTick(() => { newData = adding = null })"
        :after="[
          {
            icon: 'check',
            content: true,
            handler: handleAdd
          }
        ]"
      />
    </q-card-actions>

    methods: {
      add (type) {
        this.adding = type
        this.$nextTick(() => {
          this.$refs.inputBox.focus()
        })
      }
    }

**Comment by rstoenescu** [comment]
Closing as it's fixed in v0.15.
